crates/miden-lib/asm/kernels/transaction/lib/memory.masm

```diff
@@ -806,19 +806,19 @@ end
 #! - the provided account data pointer is equal to the native account data pointer.
 export.push_ptr_to_account_stack
     # check that the account stack is not full
-    mem_load.ACCOUNT_STACK_TOP_PTR dup push.MAX_ACCOUNT_STACK_PTR
-    lt assert.err=ERR_ACCOUNT_STACK_OVERFLOW
+    mem_load.ACCOUNT_STACK_TOP_PTR dup
+    lt.MAX_ACCOUNT_STACK_PTR assert.err=ERR_ACCOUNT_STACK_OVERFLOW
```

Comment:
nit (and probably not for this PR): we should be able to use the `u32assert2 u32lt` tick here to reduce cycle count.